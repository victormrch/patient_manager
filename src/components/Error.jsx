import React from "react";

const Error = (props) => {
  const { message } = props;
  return (
    <div className="bg-red-800 text-white text-center p-3 uppercase font-bold rounded-md mb-5">
      <p>{message}</p>
    </div>
  );
};

export default Error;
