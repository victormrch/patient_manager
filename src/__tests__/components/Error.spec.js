import React from "react";
import { render, screen } from "@testing-library/react";
import Error from "../../components/Error";
import "@testing-library/jest-dom/extend-expect";

describe("Error component specs", () => {
  it("should render correctly using snapshot testing", () => {
    const message = "Test error";

    const { asFragment } = render(<Error message={message} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the message if exist message", () => {
    const message = "Test Message";

    render(<Error message={message} />);

    const element = screen.getByText(message);

    expect(element).not.toBeNull();
    expect(element).toBeInTheDocument();
    expect(element.tagName).toEqual("P");
    expect(element.textContent).toBe("Test Message");
  });
});
