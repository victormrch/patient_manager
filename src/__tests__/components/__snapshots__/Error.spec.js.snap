// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component specs should render correctly using snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="bg-red-800 text-white text-center p-3 uppercase font-bold rounded-md mb-5"
  >
    <p>
      Test error
    </p>
  </div>
</DocumentFragment>
`;
